eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -1073,7 +1073,7 @@ protected function assertSubtreeProperties(array $expectedValues, Location $loca
         $repository = $this->getRepository();
         $locationService = $repository->getLocationService();
 
-        if ($location->id === $stopId) {
+        if ($location->id == $stopId) {
```

Comment:
This was afaik not the intention of the `mixed` notation.
The intention is that it might become a hash one day if we ever switch to NoSql storage or otherwise introduce id's that are unique in a multi master backend setup.
The intention was never to explicitly allow  22 == '22'  as far as my POV goes.

So would rather vote for having the comparison strict to be able to catch bugs like this, it will still pass if we ever switch to hash based id.